packages/server/src/api/muster/muster.compliance.ts

```diff
@@ -45,29 +46,71 @@ export function toMusterIntermediateCompliance(rosters: Roster[]): IntermediateM
  * TBD later....
  * @param observations
  * @param musteringOpportunities
- * @param musterComplianceReport
+ * @param rosterForMusterCompliance
  */
-export function calculateMusterCompliance(
-  observations: RecordedObservations[],
-  musteringOpportunities: MusteringOpportunities,
-  musterComplianceReport: IntermediateMusterCompliance[],
-): MusterCompliance[] {
-  return [];
+export function calculateMusterCompliance(observations: RecordedObservations[], musteringOpportunities: MusteringOpportunities, rosterForMusterCompliance: RosterForMusterCompliance[]): MusterCompliance[] {
+  return rosterForMusterCompliance.map(roster => {
+    const edipi = roster.edipi;
+    const unitId = roster.unitId;
+    const unitMusteringOpportunities = musteringOpportunities[unitId];
+
+    if (!unitMusteringOpportunities || unitMusteringOpportunities.length === 0) {
+      return { totalMusters: 0, mustersReported: 0, musterPercent: 100, ...roster };
+    }
+
+    const totalMusters = unitMusteringOpportunities.length;
+
+    const edipiObservations = filterObservationsByEdipi(observations, edipi);
+    const mustersReported = calculateIndividualMusterCompliance(edipiObservations, unitMusteringOpportunities);
+    const musterPercent = (100 * mustersReported) / totalMusters;
+
+
+    return { totalMusters, mustersReported, musterPercent, ...roster };
+  });
+}
+
+function calculateIndividualMusterCompliance(observations: RecordedObservationTimestamp[], musteringOpportunities: MusterWindowWithStartAndEndDate[]): number {
+  let compliance = 0;
+  musteringOpportunities.forEach(musteringOpportunity => {
+    compliance += calculateCompliance(observations, musteringOpportunity.startMusterDate, musteringOpportunity.endMusterDate);
+  });
+  return compliance;
 }
 
+function filterObservationsByEdipi(observations: RecordedObservations[], edipi: string): RecordedObservationTimestamp[] {
+  return observations
+    .filter(observation => observation.edipi === edipi)
+    .map(observation => { return { timestamp: observation.timestamp }; });
+}
+
+function calculateCompliance(observationTimestamps: RecordedObservationTimestamp[], startMusterDate: Moment, endMusterDate: Moment): number {
+  let compliance = 0;
+
+  for (const obTs of observationTimestamps) {
+    const epochTimestamp = obTs.timestamp.getTime();
+    if (epochTimestamp >= startMusterDate.valueOf() && epochTimestamp <= endMusterDate.valueOf()) {
+      compliance = 1;
+      break;
+    }
+  }
+
+  return compliance;
+}
+
+
 /**
  * Represents muster compliance for an individual
  */
 export type MusterCompliance = {
   totalMusters: number;
   mustersReported: number;
   musterPercent: number;
-} & IntermediateMusterCompliance;
+} & RosterForMusterCompliance;
 
 /**
  * Represents muster compliance without any compliance calculations
```

Comment:
This is wrong. This is just a roster a subset of roster. This should be moved to the roster module.